### PR TITLE
TwoDigitYearMax breaking change

### LINKED ISSUE #31978: [Breaking change]: TwoDigitYearMax default value changed from 2029 to 2049 in Gregorian based calendars
### Description

Calendars has the [TwoDigitYearMax](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.calendar.twodigityearmax?view=net-7.0) property which defines the last year of a 100-year range that can be represented by a 2-digit year. This property is usually used when having 2-digits years which need to be translated to the 4-digits years. This property used to be defaulted inside the Gregorian based calendars to `2029` which translate 2-digit years from `00` to `29` to be `2000` to `2029` as a 4-digits year. Any year from `30` to `99` will be translated into `1930` to `1999`. The breaking change is changing the default property value from `2029` to `2049`.  Which means it translates 2-digit years from `00` to `49` to be `2000` to `2049` as a 4-digits year. Any year from `50` to `99` will be translated into `1950` to `1999`. 

Also, when running on Windows, we'll start to get the default value of the `TwoDigitYearMax` property from Windows settings as we are doing that on .NET Framework. Also, we used to read Windows settings on NET Core up to and including version `3.1` till we switched to use ICU in .NET 5.0. Date parsing is the functionality which is the most affected by this change.

https://github.com/dotnet/runtime/issues/75148
https://github.com/dotnet/runtime/pull/76848

### Version

.NET 8 Preview 1

### Previous behavior

If not overriding the value of this property and trying to parse a date like `12/25/35` with Gregorian calendar, it used to produce the date `December 12, 1935`.  This behavior will be observed on .NET 6.0 and 7.0.

### New behavior

After the change parsing dates like `12/25/35` will produce the date `December 12, 2035`. This new behavior will be observed starting of .NET 8.0.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load/execute or different run-time behavior.
- [ ] **Source incompatible**: Source code may encounter a breaking change in behavior when targeting the new runtime/component/SDK, such as compile errors or different run-time behavior.

### Reason for change

We are already in the year `2022`. It makes more sense when parsing 2-digit years closer to current year will produce 4-digit years in current century and not the previous one. Windows OS changed its default settings too to same number `2049`.

### Recommended action

Any app not desiring to depend on the default settings when parsing and want to control the 2-digit year translation to 4-digit year, can do that by setting the desired value to [TwoDigitYearMax](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.calendar.twodigityearmax?view=net-7.0).

```C#
CultureInfo clonedInvariantCulture = (CultureInfo)(CultureInfo.InvariantCulture.Clone());
clonedInvariantCulture.DateTimeFormat.Calendar.TwoDigitYearMax = 2039; // any desired cutoff value

DateTime dt = DateTime.Parse("12/25/35", clonedInvariantCulture);
```

### Feature area

Globalization

### Affected APIs

Calendars and Date parsing APIs.

---
[Associated WorkItem - 60878](https://dev.azure.com/msft-skilling/Content/_workitems/edit/60878)

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #31978 

[Preview](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/globalization/8.0/twodigityearmax-default?branch=pr-en-us-33718)

### CODE REVIEW

File: docs/core/compatibility/globalization/8.0/twodigityearmax-default.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: "Breaking change: TwoDigitYearMax default is 2049"
+description: Learn about the globalization breaking change in .NET 8 where the default value of the GregorianCalendar.TwoDigitYearMax property has changed from 2029 to 2049.
+ms.date: 01/25/2023
+---
+# TwoDigitYearMax default is 2049
+
+Calendar classes, such as <xref:System.Globalization.GregorianCalendar>, have a `TwoDigitYearMax` property that defines the last year of a 100-year range that can be represented by a two-digit year. This property is often used to translate a two-digit year to a four-digit year. Previously, <xref:System.Globalization.GregorianCalendar.TwoDigitYearMax?displayProperty=nameWithType> defaulted to 2029. That value meant that two-digit years from 00 to 29 translated to 2000-2029. Two-digit years from 30 to 99 translated to 1930-1999. The default `TwoDigitYearMax` property value for <xref:System.Globalization.GregorianCalendar> has now changed from 2029 to 2049. The new value means that two-digit years from 00 to 49 are translated to 2000-2049. Any year from 50 to 99 will be translated to 1950-1999.
```
Comment (gewarren) [inline docs/core/compatibility/globalization/8.0/twodigityearmax-default.md:8]: Are there any other calendar classes this applies to?

File: docs/core/compatibility/globalization/8.0/twodigityearmax-default.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```csharp
+CultureInfo clonedInvariantCulture = (CultureInfo)(CultureInfo.InvariantCulture.Clone());
+clonedInvariantCulture.DateTimeFormat.Calendar.TwoDigitYearMax = 2039; // Use any desired cutoff value.
+
+DateTime dt = DateTime.Parse("12/25/45", clonedInvariantCulture);
+```
+
+## Affected APIs
+
+- <xref:System.DateTime.Parse%2A?displayProperty=fullName>
```
Comment (gewarren) [inline docs/core/compatibility/globalization/8.0/twodigityearmax-default.md:47]: I'm not sure if this is the complete list?

File: docs/core/compatibility/globalization/8.0/twodigityearmax-default.md (hunk 1 above)
Comment (tarekgh) [reply docs/core/compatibility/globalization/8.0/twodigityearmax-default.md:8]: Yes it does. It does apply on calendars which are based on Gregorian calendar like EastAsianLunisolarCalendar. Also it apply to calendar similar to Gregorian like JulianCalendar

File: docs/core/compatibility/globalization/8.0/twodigityearmax-default.md (hunk 2 above)
Comment (tarekgh) [reply docs/core/compatibility/globalization/8.0/twodigityearmax-default.md:47]: Only missing DateOnly parse methods.

**Review by tarekgh — APPROVED** [review]
LGTM, I replied to your comments. 

thanks @gewarren

**Pull request merged**